components/pango_context/src/make_drawable.cpp

```diff
@@ -143,87 +143,76 @@ Shared<Drawable> DrawableConversionTraits<draw::Icosphere>::makeDrawable(
 
-    h_angle1 += kHAngle;
-    h_angle2 += kHAngle;
-  }
+      Eigen::Vector3f mid01 = midpoint(v0, v1);
+      Eigen::Vector3f mid12 = midpoint(v1, v2);
+      Eigen::Vector3f mid20 = midpoint(v2, v0);
+
+      unsigned int i3 = static_cast<unsigned int>(vertices.size());
+      vertices.push_back(mid01);
+
+      unsigned int i4 = static_cast<unsigned int>(vertices.size());
+      vertices.push_back(mid12);
+
+      unsigned int i5 = static_cast<unsigned int>(vertices.size());
+      vertices.push_back(mid20);
 
-  vertices.push_back({0, 0, -sphere.radius});
-
-  faces.push_back(FaceT(0, 1, 3));
-  faces.push_back(FaceT(0, 3, 5));
-  faces.push_back(FaceT(0, 5, 7));
-  faces.push_back(FaceT(0, 7, 9));
-  faces.push_back(FaceT(0, 9, 1));
-
-  faces.push_back(FaceT(11, 4, 2));
-  faces.push_back(FaceT(11, 6, 4));
-  faces.push_back(FaceT(11, 8, 6));
-  faces.push_back(FaceT(11, 10, 8));
-  faces.push_back(FaceT(11, 2, 10));
-
-  faces.push_back(FaceT(1, 2, 3));
-  faces.push_back(FaceT(2, 4, 3));
-  faces.push_back(FaceT(3, 4, 5));
-  faces.push_back(FaceT(4, 6, 5));
-  faces.push_back(FaceT(5, 6, 7));
-  faces.push_back(FaceT(6, 8, 7));
-  faces.push_back(FaceT(7, 8, 9));
-  faces.push_back(FaceT(8, 10, 9));
-  faces.push_back(FaceT(9, 10, 1));
-  faces.push_back(FaceT(10, 2, 1));
-
-  for (size_t i = 0; i < sphere.num_subdivisions; ++i) {
-    std::vector<FaceT> source_faces = faces;
-    faces.clear();
-    for (const FaceT& t : source_faces) {
-      Eigen::Vector3f p0_p1 =
-          (vertices[t.x()] + vertices[t.y()]).normalized() * sphere.radius;
-      int p0_p1_id = vertices.size();
-      vertices.push_back(p0_p1);
-
-      Eigen::Vector3f p0_p2 =
-          (vertices[t.x()] + vertices[t.z()]).normalized() * sphere.radius;
-      int p0_p2_id = vertices.size();
-      vertices.push_back(p0_p2);
-
-      Eigen::Vector3f p1_p2 =
-          (vertices[t.y()] + vertices[t.z()]).normalized() * sphere.radius;
-      int p1_p2_id = vertices.size();
-      vertices.push_back(p1_p2);
-
-      faces.push_back(FaceT(t.x(), p0_p1_id, p0_p2_id));
-      faces.push_back(FaceT(t.y(), p1_p2_id, p0_p1_id));
-      faces.push_back(FaceT(t.z(), p0_p2_id, p1_p2_id));
-      faces.push_back(FaceT(p0_p1_id, p1_p2_id, p0_p2_id));
+      new_indices.push_back(i0);
+      new_indices.push_back(i3);
+      new_indices.push_back(i5);
+
+      new_indices.push_back(i1);
+      new_indices.push_back(i4);
+      new_indices.push_back(i3);
+
+      new_indices.push_back(i2);
+      new_indices.push_back(i5);
+      new_indices.push_back(i4);
+
+      new_indices.push_back(i3);
+      new_indices.push_back(i4);
+      new_indices.push_back(i5);
     }
+
+    indices.swap(new_indices);
   }
 
+  auto colors = std::vector<Color>(vertices.size(), sphere.color);
+
   prims->vertices->queueUpdate(std::move(vertices));
-  prims->indices->queueUpdate(std::move(faces));
+  prims->indices->queueUpdate(std::move(indices));
+  prims->colors->queueUpdate(std::move(colors));
```

Comment:
Rather than add a vbo of duplicate entries like this, you should be able to just set default_color (https://github.com/stevenlovegrove/Pangolin/blob/pango2/components/pango_context/include/pangolin/drawable/drawn_primitives.h#L112)

Reply:
thanks, done